runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -325,6 +325,20 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
                 new StreamingFnApiCreateOverrideFactory()));
       }
       overridesBuilder
+          // Support Splittable DoFn for now only in streaming mode.
+          // The order of the following overrides is important because they are applied in order.
+
+          // Dataflow runner does not expand single-output ParDo into multi-output ParDo
+          // by default. However, we *do* want to do that for both single-output splittable ParDo.
```

Comment:
This is specifically required because if it is not performed, the enclosing node will be replaced at a later point, effectively "losing" the application of the later override.